apps/fortuna/src/keeper.rs

```diff
@@ -297,41 +270,64 @@ pub async fn process_block_range(
         if to_block > last_block {
             to_block = last_block;
         }
+
+        process_block_batch(
+            BlockRange {
+                from: current_block,
+                to:   to_block,
+            },
+            contract.clone(),
+            gas_limit,
+            chain_state.clone(),
+        )
+        .in_current_span()
+        .await;
+
+        current_block = to_block + 1;
+    }
+}
+
+/// Process a batch of blocks for a chain. It will fetch events for the blocks for the provided batch
```

Comment:
You can specify that it's gonna try to get all the events using a single call to distinguish this function and the other one.